File: admin-dashboard/src/api/vip.ts
import { api } from './client';

export interface VipLevelItem {
  level: number;
  min_spend: number;
  min_visits: number;
  benefit: string;
  is_active: boolean;
}

export interface VipLevelsUpdatePayload {
  levels: VipLevelItem[];
}

export const getAdminVipLevels = async () => {
  const response = await api.get('/vip/admin/levels');
  return (response.data || []) as VipLevelItem[];
};

export const updateAdminVipLevels = async (payload: VipLevelsUpdatePayload) => {
  const response = await api.put('/vip/admin/levels', payload);
  return (response.data || []) as VipLevelItem[];
};

File: backend/app/schemas/vip.py
"""
VIP schemas
"""
from typing import Optional
from pydantic import BaseModel


class VipLevelItem(BaseModel):
    level: int
    min_spend: float
    min_visits: int
    benefit: str
    is_active: bool = True


class VipLevelUpsertItem(BaseModel):
    level: int
    min_spend: float
    min_visits: int
    benefit: str
    is_active: bool = True


class VipLevelsUpdateRequest(BaseModel):
    levels: list[VipLevelUpsertItem]


class VipProgress(BaseModel):
    current: float
    required: float
    percent: float


class VipStatusResponse(BaseModel):
    current_level: VipLevelItem
    total_spend: float
    total_visits: int
    spend_progress: VipProgress
    visits_progress: VipProgress
    next_level: Optional[VipLevelItem] = None
